### PR TITLE
feat(frontend): add badge component

### PR DESCRIPTION
**Pull request by loki344**
# Motivation

Implement the reusable component 'Badge' from figma.

image

# Changes

- Implement Badge
- Replace current concrete 'implementations' in DappTags with abstracted component

# Tests

Looks same same:

new: 
image
image

### CODE REVIEW

**Comment by peterpeterparker** [comment]
Couldn't we use the component of the UI kit https://gix.design/components/tag ?

**Comment by loki344** [comment]
Yes would be great! I'll check the possibilities to adjust the colors to match the figma design

**Comment by loki344** [comment]
So to match the gix tag to the figma design:
- I have to override basically all color variables from gix for the tag
- I have to implement the missing variants: Outline and Default

I would rather have a own component, but ultimately I let you decide what is best for this project.

File: src/frontend/src/lib/components/ui/Badge.svelte [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+		default: 'border border-light-grey bg-dust/30',
+		info: 'bg-light-blue text-blue-ribbon',
+		error: 'bg-rusty-red/20 text-rusty-red',
+		warning: 'bg-american-orange/20 text-american-orange',
+		success: 'bg-green-crayola/20 text-green-crayola',
+		outline: 'border border-light-grey bg-off-white'
+	};
+</script>
+
+<div
```
Comment (peterpeterparker) [inline src/frontend/src/lib/components/ui/Badge.svelte:15]: Shouldn't it be either a `span` or `li`?

File: src/frontend/src/lib/components/dapps/DappTags.svelte [hunk 2]
```diff
@@ -11,8 +12,8 @@
 	class="flex list-none flex-wrap gap-2"
 >
 	{#each tags as tag}
-		<li class="rounded border border-light-grey bg-dust/30 px-2 py-0.5 text-xs font-semibold">
+		<Badge>
```
Comment (peterpeterparker) [inline src/frontend/src/lib/components/dapps/DappTags.svelte:15]: This should have been another PR. Anyway, see other comment, it's not a `li` anymore.

File: src/frontend/src/lib/components/ui/Badge.svelte [hunk 3]
```diff
@@ -0,0 +1,21 @@
+<script lang="ts">
+	export let variant: 'default' | 'info' | 'error' | 'warning' | 'success' | 'outline' = 'default';
```
Comment (peterpeterparker) [inline src/frontend/src/lib/components/ui/Badge.svelte:2]: There isn't a better way to type this in TypeScript in order to exactly reflects the possible types of the `variableClassNames`?

**Comment by peterpeterparker** [comment]
> I would rather have a own component
Ok

File: src/frontend/src/lib/components/ui/Badge.svelte (hunk 1 above)
Comment (loki344) [reply src/frontend/src/lib/components/ui/Badge.svelte:15]: I'll make it a span

File: src/frontend/src/lib/components/ui/Badge.svelte (hunk 3 above)
Comment (loki344) [reply src/frontend/src/lib/components/ui/Badge.svelte:2]: I changed it to: `export let variant: keyof typeof variantClassNames = 'default';`

File: src/frontend/src/lib/components/ui/Badge.svelte [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+		error: 'bg-rusty-red/20 text-rusty-red',
+		warning: 'bg-american-orange/20 text-american-orange',
+		success: 'bg-green-crayola/20 text-green-crayola',
+		outline: 'border border-light-grey bg-off-white'
+	};
+
+	export let variant: keyof typeof variantClassNames = 'default';
+</script>
+
+<span
```
Comment (peterpeterparker) [inline src/frontend/src/lib/components/ui/Badge.svelte:16]: Forgot to ask, one last question: ensuring text are trimmed if too long is repsonsability of consumer of the badge?

Comment (loki344) [reply src/frontend/src/lib/components/ui/Badge.svelte:16]: Good point. I could add a text-ellipsis but would expect the consumer to define the width of the badge?

Comment (peterpeterparker) [reply src/frontend/src/lib/components/ui/Badge.svelte:16]: Yeah I'm not sure neither, that's why I asked ;)

Maybe have a try with your current feature, what happen if a `tag` is super long?

Note: I'll approve the PR that's something we can improve in another PR.

**Review by peterpeterparker — APPROVED** [review]
LGTM, thx

**Pull request merged**